Source/JavaScriptCore/heap/Heap.cpp

```diff
@@ -3328,4 +3328,61 @@ DEFINE_DYNAMIC_ISO_SUBSPACE_MEMBER_SLOW(moduleProgramExecutableSpace)
 
 } // namespace GCClient
 
+#if ASSERT_ENABLED
+static Lock strongMapLock;
+static HashMap<uintptr_t, String>* debugStrongMap;
+static std::atomic<int> strongLoggingDisabled;
+static std::atomic<uintptr_t> lastStrongID;
+
+StrongLoggingDisabledScope::StrongLoggingDisabledScope()
+{
+    if (!Options::logLiveStrongs())
+        return;
+    ++strongLoggingDisabled;
+}
+
+StrongLoggingDisabledScope::~StrongLoggingDisabledScope()
+{
+    if (!Options::logLiveStrongs())
+        return;
+    --strongLoggingDisabled;
+}
+
+uintptr_t debugReportLiveStrong()
+{
+    if (!Options::logLiveStrongs() || strongLoggingDisabled.load() > 0)
+        return 0;
+    Locker locker(strongMapLock);
+    if (!debugStrongMap)
+        debugStrongMap = new HashMap<uintptr_t, String>();
+    WTF::StringPrintStream os;
+    WTFReportBacktraceWithPrefixAndPrintStream(os, "");
```

Comment:
Can you use `WTF::StackShot` instead of a string?

Then the map is `HashMap<token, std::unique_ptr<StackShot>>`